src/core/automount.c

```diff
@@ -1038,18 +1037,6 @@ static int automount_dispatch_io(sd_event_source *s, int fd, uint32_t events, vo
                         goto fail;
                 }
 
-                /* Before we do anything, let's see if somebody is playing games with us? */
-                if (lstat(a->where, &st) < 0) {
-                        log_unit_warning_errno(UNIT(a), errno, "Failed to stat automount point: %m");
-                        goto fail;
-                }
-
-                if (!S_ISDIR(st.st_mode) || st.st_dev == a->dev_id) {
-                        log_unit_info(UNIT(a), "Automount point already unmounted?");
-                        automount_send_ready(a, a->expire_tokens, 0);
-                        break;
-                }
-
```

Comment:
statx doesn't help - partly because it always revalidates the path that you pass (quite independent of the request mask) and partly because AT_STATX_DONT_SYNC is currently unimplemented, and you don't even get an error telling you that it isn't implemented (sad!).
There really is no safe way for pid-1 to stat any file that might be on NFS.  None at all.
Possibly the current functionality could be duplicated safely by parsing /proc/self/mountinfo, but I don't see the point.  What *is* the reason that code is there?  Where is the harm in calling umount if the filesystems happened to have already been unmount?